routes/base.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module CloverBase
+  def self.included(base)
+    base.extend(ClassMethods)
+    base.plugin :all_verbs
+    base.plugin :request_headers
+
+    logger = if ENV["RACK_ENV"] == "test"
+      Class.new {
+        def write(_)
+        end
+      }.new
+    else
+      # :nocov:
+      $stderr
+      # :nocov:
+    end
+    base.plugin :common_logger, logger
+
+    # :nocov:
+    case Config.mail_driver
+    when :smtp
+      ::Mail.defaults do
+        delivery_method :smtp, {
+          address: Config.smtp_hostname,
+          port: Config.smtp_port,
+          user_name: Config.smtp_user,
+          password: Config.smtp_password,
+          authentication: :plain,
+          enable_starttls: Config.smtp_tls
+        }
+      end
+    when :logger
+      ::Mail.defaults do
+        delivery_method :logger
+      end
+    when :test
+      ::Mail.defaults do
+        delivery_method :test
+      end
+    end
+    # :nocov:
+  end
+
+  # TODO(enes): Handle exception better way, maybe bind http codes with exceptions
```

Comment:
```suggestion
  # Assign some HTTP response codes to common exceptions.
```

I think this TODO so terse we'll (or at least many of us) wonder what the heck we were talking about in a few months. I think a reasonable software designer can come up with the idea of in-situ exceptions-with-http-codes on their own (although they may wonder if it's a good idea, to which I think, with some experience, I would say: yeah probably).

If you really want to emphasize the point, in a follow-up, find some spots where you want a nice error message and then implement (and feel out) the pattern at your earliest convenience. Sometimes to show is really better to describe.

Reply:
I updated comment. I will show some example in future probably